ci/requirements/min-all-deps.yml

```
@@ -7,33 +7,33 @@ dependencies:
   # Run ci/min_deps_check.py to verify that this file respects the policy.
   # When upgrading python, numpy, or pandas, must also change
   # doc/user-guide/installing.rst, doc/user-guide/plotting.rst and setup.py.
-  - python=3.8
+  - python=3.9
   - boto3=1.20
   - bottleneck=1.3
   - cartopy=0.20
   - cdms2=3.1
   - cfgrib=0.9
   - cftime=1.5
   - coveralls
-  - dask-core=2021.11
-  - distributed=2021.11
+  - dask-core=2022.1
+  - distributed=2022.1
   - flox=0.5
-  - h5netcdf=0.11
+  - h5netcdf=0.13
   # h5py and hdf5 tend to cause conflicts
   # for e.g. hdf5 1.12 conflicts with h5py=3.1
   # prioritize bumping other packages instead
   - h5py=3.6
   - hdf5=1.12
   - hypothesis
   - iris=3.1
-  - lxml=4.6  # Optional dep of pydap
+  - lxml=4.7  # Optional dep of pydap
   - matplotlib-base=3.5
   - nc-time-axis=1.4
   # netcdf follows a 1.major.minor[.patch] convention
   # (see https://github.com/Unidata/netcdf4-python/issues/1090)
   - netcdf4=1.5.7
-  - numba=0.54
-  - numpy=1.20
+  - numba=0.55
+  - numpy=1.21
   - packaging=21.3
   - pandas=1.3
```

to make #7373 a bit easier, let's also bump `pandas`:
```suggestion
  - pandas=1.4
```